### DISCUSSION TITLE
Label and Component focus

### DISCUSSION THREAD

**Discussion by RAN-Vestjylland** [BlazorFluentUI/BlazorFluentUI#306] [Ideas]
`<TextFieldNumber Label="Alder" @bind-Value="PartnerModel.Age" />`
Gives this with automatical focus on input field when clicking on the label:
![2021-05-03 11_58_13-VES TBS3](https://user-images.githubusercontent.com/78351622/116863641-e1064d80-ac06-11eb-95a1-1c00822f5419.png)

and the following
`<Label>Alder</Label>`
`<TextFieldNumber @bind-Value="PartnerModel.Age" />`
Gives this where you can't click the label to set focus on the input field:
![2021-05-03 12_01_17-VES TBS3](https://user-images.githubusercontent.com/78351622/116863920-4b1ef280-ac07-11eb-8927-76804ed5920e.png)

But by for instance, by having For and ForID it would be able to place the Label control anywhere you want (like in normal html) and make it focus on a specific component when clicking on the text of the label:
`<Label For="PartnerModelAgeInput">Alder</Label>`
`<TextFieldNumber ForID="PartnerModelAgeInput" @bind-Value="PartnerModel.Age" />`

The reason I suggest this, is because the ID of the input field are a random generated value. It really doesn't matter as it is, but in some instances, like mine, it makes sense to actually be able to specifically target that component directly. This means that it would also be possible to place labels whereever the design teams feels like and for instance follow the design of a specific unmentioned CRM system 😃

**Comment by limefrogyank** [comment]
We try to keep it simple by emulating the react components that Microsoft made... and this doesn't have that.  However, being React components, you could really add anything you want.  

The equivalent here would be to use attribute splatting.  
https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-5.0#attribute-splatting-and-arbitrary-parameters

However, I don't think we have this built-in to most of our components.  If you don't break any existing functionality, feel free to add this yourself.  Then I can see your scenario being doable.
